app/src/main/java/com/example/travenor/screen/search/SearchFragment.kt

```diff
@@ -0,0 +1,218 @@
+package com.example.travenor.screen.search
+
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import androidx.annotation.RequiresApi
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.ViewCompat
+import com.example.travenor.data.model.photo.PlacePhoto
+import com.example.travenor.data.model.place.Place
+import com.example.travenor.data.repository.PlaceRepositoryImpl
+import com.example.travenor.data.source.local.PlaceExploreLocalSource
+import com.example.travenor.data.source.local.PlaceLocalDataSource
+import com.example.travenor.data.source.remote.PlaceRemoteDataSource
+import com.example.travenor.databinding.FragmentSearchBinding
+import com.example.travenor.screen.detail.DetailActivity
+import com.example.travenor.screen.search.adapter.RecentSearchAdapter
+import com.example.travenor.screen.search.adapter.SearchResultAdapter
+import com.example.travenor.utils.base.BaseFragment
+
+@Suppress("TooManyFunctions")
+class SearchFragment :
+    BaseFragment<FragmentSearchBinding>(),
+    SearchContract.View,
+    RecentSearchAdapter.OnItemClickListener,
+    SearchResultAdapter.OnItemClickListener {
+
+    private val mRecentSearchAdapter: RecentSearchAdapter by lazy { RecentSearchAdapter() }
+    private val mSearchResultAdapter: SearchResultAdapter by lazy { SearchResultAdapter() }
+    private var mPresenter: SearchPresenter? = null
+
+    override fun inflateViewBinding(inflater: LayoutInflater): FragmentSearchBinding {
+        return FragmentSearchBinding.inflate(inflater)
+    }
+
+    private fun setLayoutBelowSystemBar() {
```

Comment:
Các private func em move vào cuối file nhé,
Thống nhất cho dễ đọc code sau này!

sx ntn:
- public method
- Override func of parent class
- Override func interface implement
- private method
- ....

Reply:
Vâng a